File: src/main/java/com/freya02/botcommands/api/localization/DefaultLocalizationMap.java
package com.freya02.botcommands.api.localization;

import org.jetbrains.annotations.NotNull;

import java.util.Locale;
import java.util.Map;

/**
 * POJO for effective locale and localization template map
 */
public record DefaultLocalizationMap(Locale effectiveLocale,
                                     Map<String, ? extends LocalizationTemplate> templateMap) implements LocalizationMap {
	@Override
	@NotNull
	public Locale effectiveLocale() {
		return effectiveLocale;
	}

	@Override
	@NotNull
	public Map<String, ? extends LocalizationTemplate> templateMap() {
		return templateMap;
	}
}

File: src/main/kotlin/com/freya02/botcommands/api/localization/DefaultLocalizationMap.kt
package com.freya02.botcommands.api.localization

import java.util.*

class DefaultLocalizationMap(
    override val effectiveLocale: Locale,
    private val localizationMap: Map<String, LocalizationTemplate?>
) : LocalizationMap {
    override fun get(path: String): LocalizationTemplate? = localizationMap[path]

    override fun toString(): String {
        return "DefaultLocalizationMap(effectiveLocale=$effectiveLocale)"
    }
}
